expr/func_summarize_test.go

```diff
@@ -665,6 +665,78 @@ func TestSummarizeAlignToFrom(t *testing.T) {
 	testSummarize("AlignToFrom", input, outputMax[1], "45s", "max", true, t)
 }
 
+func TestSummarizeLargeIntervalTimestamps(t *testing.T) {
+
+	// This test is specifically testing the timestamp behavior, so we don't need real values.
+	// However, we do need a lot of datapoints to trigger the bug we are regression testing against.
+	interval1Day := uint32(24 * 60 * 60)
+	numTimestamps := uint32(2000)
+	startTime := uint32(1464637518)
+	endTime := startTime + numTimestamps*interval1Day
+
+	inputDps := make([]schema.Point, 0, numTimestamps)
+
+	for i := uint32(0); i < numTimestamps; i++ {
+		inputDps = append(inputDps, schema.Point{Val: 0, Ts: startTime + i*interval1Day})
+	}
+
+	outputInterval := 365 * interval1Day
+
+	var unaligned365dsum = []schema.Point{
+		{Val: 0, Ts: 1450656000},
+		{Val: 0, Ts: 1482192000},
+		{Val: 0, Ts: 1513728000},
+		{Val: 0, Ts: 1545264000},
+		{Val: 0, Ts: 1576800000},
+		{Val: 0, Ts: 1608336000},
+	}
+
+	var aligned365dsum = []schema.Point{
+		{Val: 0, Ts: 1464637518},
+		{Val: 0, Ts: 1496173518},
+		{Val: 0, Ts: 1527709518},
+		{Val: 0, Ts: 1559245518},
+		{Val: 0, Ts: 1590781518},
+		{Val: 0, Ts: 1622317518},
```

Comment:
what's the relation between these ts? can they be set from code so it's more obvious?